Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm

```diff
@@ -2410,11 +2411,67 @@ static IntRect computeMarqueeSelectionRect(const WebCore::IntPoint& point1, cons
     return true;
 }
 
-bool UnifiedPDFPlugin::performDictionaryLookupAtLocation(const FloatPoint&)
+bool UnifiedPDFPlugin::performDictionaryLookupAtLocation(const FloatPoint& rootViewPoint)
 {
+    IntPoint pluginPoint = convertFromRootViewToPlugin(roundedIntPoint(rootViewPoint));
+    IntPoint documentPoint = convertFromPluginToDocument(pluginPoint);
+    auto pageIndex = pageIndexForDocumentPoint(documentPoint);
+    if (!pageIndex)
+        return false;
+
+    auto page = m_documentLayout.pageAtIndex(*pageIndex);
+    RetainPtr lookupSelection = [page selectionForWordAtPoint:convertFromDocumentToPage(documentPoint, *pageIndex)];
+    return searchInDictionary(WTFMove(lookupSelection));
+}
+
+bool UnifiedPDFPlugin::searchInDictionary(const RetainPtr<PDFSelection>& lookupSelection)
+{
+    RetainPtr scaledString = [lookupSelection attributedStringScaled:(m_scaleFactor * m_documentLayout.scale())];
+
+    if (auto selectionBounds = selectionBoundsForFirstPageInDocumentSpace(lookupSelection))
+        return showDefinitionForAttributedString(WTFMove(scaledString), *selectionBounds);
+
     return false;
 }
 
+std::optional<IntRect> UnifiedPDFPlugin::selectionBoundsForFirstPageInDocumentSpace(const RetainPtr<PDFSelection>& selection) const
+{
+    if (!selection)
+        return std::nullopt;
+
+    for (PDFPage *page in [selection pages]) {
+        auto pageIndex = m_documentLayout.indexForPage(page);
+        if (!pageIndex)
+            continue;
+        auto rectForPage = IntRect { [selection boundsForPage:page] };
+        return convertFromPageToDocument(rectForPage, *pageIndex);
+    }
+
+    return std::nullopt;
+}
+
+bool UnifiedPDFPlugin::showDefinitionForAttributedString(RetainPtr<NSAttributedString>&& string, const IntRect& rectInDocumentSpace)
+{
+    auto rectInRootView = convertFromPluginToRootView(convertFromDocumentToPlugin(rectInDocumentSpace));
+
+    DictionaryPopupInfo dictionaryPopupInfo;
+    dictionaryPopupInfo.origin = rectInRootView.location();
+    dictionaryPopupInfo.platformData.attributedString = WebCore::AttributedString::fromNSAttributedString(string.get());
+
+    TextIndicatorData dataForSelection;
+    dataForSelection.selectionRectInRootViewCoordinates = rectInRootView;
+    dataForSelection.textBoundingRectInRootViewCoordinates = rectInRootView;
+    dataForSelection.contentImageScaleFactor = m_scaleFactor * m_documentLayout.scale();
```

Comment:
Weird to set this when you don't have a contentImage but maybe it gets used for something else?

Reply:
Yes, this is not necessary. I was following precedence set by PDFPlugin::showDefinitionForAttributedString and WebPage::dictionaryPopupInfoForSelectionInPDFPlugin. I'll drop it for now with the assumption that it's redundant.